src/cow-react/modules/limitOrders/updaters/QuoteUpdater/hooks/useFetchMarketPrice.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect } from 'react'
+import { useWeb3React } from '@web3-react/core'
+import { useUpdateAtom } from 'jotai/utils'
+
+import { getQuote } from '@cow/api/gnosisProtocol'
+import { useLimitOrdersTradeState } from '@cow/modules/limitOrders/hooks/useLimitOrdersTradeState'
+import { updateLimitRateAtom } from '@cow/modules/limitOrders/state/limitRateAtom'
+import { useQuoteRequestParams } from './useQuoteRequestParams'
+import { useHandleResponse } from './useHandleResponse'
+import { useHandleError } from './useHandleError'
+
+// Every 10s
+const REFETCH_CHECK_INTERVAL = 10000
+
+export function useFetchMarketPrice() {
```

Comment:
I think it makes sense to move the hook in the directory:
`hooks/useFetchMarketPrice/index.ts`
And also move `useHandleError` and `useHandleResponse` to the directory